File: Ruby/6 KYU/Build a pile of Cubes.rb
def find_nb(m)
  n = 0
  loop do
    if m > 0
      current_cube_volume = (n+1) ** 3
      m -= current_cube_volume
    elsif m == 0
      return n
    elsif m < 0
      return -1
    end
    n += 1
  end
end

File: Javascript/6 KYU/Build a pile of Cubes.js
function findNb(m) {
  for (let n = 0;;n++) {
    if (m > 0) {
      let current_cube_volume = Math.pow(n+1,3);
      m = m - current_cube_volume;
    } else if (m == 0) {
      return n;
    } else if (m < 0) {
      return -1;
    }
  }  
}
